src/client/Client.cc

```diff
@@ -6524,13 +6524,17 @@ int Client::_do_setattr(Inode *in, struct stat *attr, int mask, int uid, int gid
       mask &= ~(CEPH_SETATTR_MTIME|CEPH_SETATTR_ATIME);
     }
   }
-  if (!mask)
+  if (!mask) {
+    in->change_attr++;
     return 0;
+  }
 
 force_request:
   MetaRequest *req = new MetaRequest(CEPH_MDS_OP_SETATTR);
 
   filepath path;
+
+  in->change_attr++;
```

Comment:
In this branch we're doing an MDS request to setattr, and handle_client_setattr bumps the change_attr — or I guess we don't include the change_attr on regular requests, just on cap messages, so bumping it on both sides merely keeps them in sync?

But if the MDS rejects the setattr request for any reason, then we've incorrectly bumped it locally. And we fill in our trace from the MDS on a setattr, so we'll update it from the MDS anyway, right? So I think we don't want to bump it ourselves if we're delegating the actual change to the MDS. Does that make sense?